src/components/SlateEditor/plugins/PluginFactory.tsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { Editor, NodeEntry, Element } from "slate";
+import { RenderLeafProps } from "slate-react";
+import config from "../../../config";
+import { NormalizerConfig, defaultBlockNormalizer } from "../utils/defaultNormalizer";
+
+type ElementType = Element["type"];
+
+type MappedPlugins = {
+  [K in ElementType]: Plugin<K>;
+};
+
+interface Plugin<T extends ElementType = ElementType> {
+  /**
+   * The unique identifier for the element in the slate context.
+   */
+  type: T;
+
+  /**
+   * Array of normalization objects containing a description and a normalisation method.
+   */
+  normalizeMethods?: Normalize<Extract<Element, { type: T }>>[];
+
+  /**
+   * Configuration object for the default block normalizer. If this is set, the default block normalizer will be used.
+   */
+  normalizeWithConfig?: NormalizerConfig;
+
+  /**
+   * onKeyDown is used to handle keyboard events for the element. Matches the string literal to the corresponding KeyboardEvent key.
+   */
+  onKeyDown?: Record<string, KeyDown<T>>;
+
+  /**
+   * Controls if the element should be considered an inline element in the slate context.
+   * Slate defines Inline elements as elements that can place within a text node. In our use case a paragraph element.
+   */
+  isInline?: boolean;
+
+  /**
+   * Controls if the element should be considered a void element in the slate context.
+   * A Void element is an element that can not have any children nodes as children, but do require a [{ text: "" }] array as a child for pathing purposes.
+   */
+  isVoid?: boolean;
+
+  /**
+   *  A method to control if the BlockPickerTrigger should be shown or not given the selection is in a node for this plugin.
+   */
+  shouldHideBlockPicker?: (editor: Editor) => boolean;
```

Comment:
Er litt usikker på om jeg liker navnet på denne, synes det fort ser mer komplisert og omstendelig ut når attributter, variabler og lignende begynner på ord som `should`. Personlig ville jeg gått for `showBlockPicker` med defaultverdi `true`, altså motsatt tankegang. Men `hideBlockPicker` vil også fungere (så logikk ellers ikke må snus) og vil være kortere. Og, imo, hakket mer lesbart når det begynner rett på `hide` eller `show`.